### DISCUSSION TITLE
Dataset.records (.tables)?

### DISCUSSION THREAD

**Discussion by rabitt** [mir-dataset-loaders/mirdata#458] [API Design]
For top-level metadata, sql tables, and other "cross-track" data, we've talked about supporting a `Dataset.records` object as a compliment to `Dataset.tracks` and `Dataset.multitracks`

Some things to decide/discuss:
* What should the API look like?
* How should we support cross record queries (e.g. sql joins)
* Can we abstract existing "metadata" into records?

**Comment by nkundiushuti** [comment]
I had two meetings with Andres Ferraro & Sergio Oramas to talk about this implementation:
 - What file formats do we take as data input: csv, tsv, json 
 - Implementation: 
a. iterator (return records-tuples) 
b. return a data-frame (load the table with pandas and return it) 
 - what problems we may face: data is too large to be fit into memory; solution: generator
 - what can we do to ease access: Helper functions - examples of data loading : split on time (train/test) 
 - MTG datasets we want to implement 
a. https://www.upf.edu/web/mtg/mumu 
b. https://mtg.github.io/melon-music-dataset/ 
c. https://www.upf.edu/web/mtg/lastfm360k
